pkg/pillar/zedUpload/datastore_azure.go

```diff
@@ -162,6 +164,34 @@ func (ep *AzureTransportMethod) processAzureUploadByChunks(req *DronaRequest) er
 	return azure.UploadPartByChunk(ep.acName, ep.acKey, ep.container, req.localName, req.UploadID, ep.hClient, req.Adata)
 }
 
+func (ep *AzureTransportMethod) processAzureDownloadByChunks(req *DronaRequest) error {
+	file := req.name
+	dataChan := make(utils.ChunkDataChan)
+	go func(req *DronaRequest, dataNotif utils.ChunkDataChan) {
+		for {
+			select {
+			case data, ok := <-dataNotif:
+				if !ok {
+					return
+				}
+				ep.ctx.postChunk(req, data.Size, data.Chunk)
+			}
+		}
+	}(req, dataChan)
+	err := azure.DownloadAzureBlobByChunks(ep.acName, ep.acKey, ep.container, file, req.objloc, ep.hClient, dataChan)
+	if err != nil {
+		close(dataChan)
+		return err
+	}
+	cleanUp(req, dataChan)
+	return nil
+}
+
+func cleanUp(req *DronaRequest, dataChan utils.ChunkDataChan) {
```

Comment:
This function makes it harder to understand the code.

Reply:
noted